File: modulo-java/src/com/fabrica/muebles/modelo/Administracion.java
package com.fabrica.muebles.modelo;

import com.fabrica.muebles.dao.ClienteDAO;
import com.fabrica.muebles.dao.ProveedorDAO;
import java.util.List;

public class Administracion {

    private final ClienteDAO clienteDAO = new ClienteDAO();
    private final ProveedorDAO proveedorDAO = new ProveedorDAO();

    // ============ CLIENTES ============
    public boolean agregarCliente(Cliente c) { 
        return clienteDAO.insertarCliente(c); 
    }

    public List<Cliente> listarClientes() { 
        return clienteDAO.consultarTodos(); 
    }

    public Cliente buscarCliente(int id) { 
        return clienteDAO.consultarPorId(id); 
    }

    public boolean actualizarCliente(Cliente c) { 
        return clienteDAO.actualizarCliente(c); 
    }

    public boolean eliminarCliente(int id) { 
        return clienteDAO.eliminarCliente(id); 
    }

    // ============ PROVEEDORES ============
    public boolean agregarProveedor(Proveedor p) { 
        return proveedorDAO.agregar(p); 
    }

    public List<Proveedor> listarProveedores() { 
        return proveedorDAO.listar(); 
    }

    public Proveedor buscarProveedor(int id) { 
        return proveedorDAO.buscar(id); 
    }

    public boolean actualizarProveedor(Proveedor p) { 
        return proveedorDAO.actualizar(p); 
    }

    public boolean eliminarProveedor(int id) { 
        return proveedorDAO.eliminar(id); 
    }
}

File: src/com/fabrica/muebles/modelo/administracion.py
from com.fabrica.muebles.dao.cliente_dao import ClienteDAO
from com.fabrica.muebles.dao.proveedor_dao import ProveedorDAO


class Administracion:
    """Clase que centraliza las operaciones de administración"""

    def __init__(self):
        self.cliente_dao = ClienteDAO()
        self.proveedor_dao = ProveedorDAO()

    # CLIENTES
    def agregar_cliente(self, cliente):
        return self.cliente_dao.insertar_cliente(cliente)

    def listar_clientes(self):
        return self.cliente_dao.consultar_todos()

    def buscar_cliente(self, id):
        return self.cliente_dao.consultar_por_id(id)

    def actualizar_cliente(self, cliente):
        return self.cliente_dao.actualizar_cliente(cliente)

    def eliminar_cliente(self, id):
        return self.cliente_dao.eliminar_cliente(id)

    # PROVEEDORES
    def agregar_proveedor(self, proveedor):
        return self.proveedor_dao.agregar(proveedor)

    def listar_proveedores(self):
        return self.proveedor_dao.listar()

    def buscar_proveedor(self, id):
        return self.proveedor_dao.buscar(id)

    def actualizar_proveedor(self, proveedor):
        return self.proveedor_dao.actualizar(proveedor)

    def eliminar_proveedor(self, id):
        return self.proveedor_dao.eliminar(id)